src/main/java/com/appland/appmap/util/EventUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.appland.appmap.util;
+
+import com.appland.appmap.output.v1.Event;
+import com.appland.appmap.transform.annotations.MethodEvent;
+
+public class EventUtil {
+
+    /**
+     *
+     * @param event
+     * @return true is the event is of type return
+     */
+    public static boolean isReturnEvent(Event event) {
+        return event.event.equalsIgnoreCase(MethodEvent.METHOD_RETURN.getEventString());
+    }
+
+    /**
+     * Removes unnecessary or duplicated fields depending on the event type
+     */
+    public static void removeUnnecessaryInfoForReturnEvents(Event event) {
+        if (event != null && event.event != null && isReturnEvent(event)) {
```

Comment:
The null checks should move into `isReturnEvent` to cover NPE in both methods. Otherwise a null event or event type will cause `isReturnEvent` to throw (which is currently happening when running this code from the Maven plugin tests).